A4A > Sites: Implement site connection error preview

yashwin opened the pull request:
Closes https://github.com/Automattic/automattic-for-agencies-dev/issues/622

## Proposed Changes

This PR implements a site error preview when a site has a connection error. 

## Testing Instructions

1. Open the A4A live link.
2. Go to Sites > Verify that the error message when there is a connection error is displayed, as shown below, with a `Fix now` button. Also, verify that the action icon(`...`) is hidden and only the `>` icon is shown.

<img width="1427" alt="Screenshot 2024-07-10 at 11 31 36 AM" src="https://github.com/Automattic/wp-calypso/assets/10586875/d450a32b-ffc0-4683-bf81-110cf474afaf">

3. Hover over the message, and it shows the tooltip as mentioned below:

When the A4A plugin is installed

<img width="1427" alt="Screenshot 2024-07-10 at 11 35 36 AM" src="https://github.com/Automattic/wp-calypso/assets/10586875/ac7e9628-699b-4ebb-86d3-1dc75a5e324d">

When the Jetpack plugin is installed


<img width="403" alt="Screenshot 2024-07-09 at 2 30 17 PM" src="https://github.com/user-attachments/assets/0b30c7cb-4b51-441f-872b-dc971d79acc5">


4. Click the `Fix now` button & the `>` icon & verify that the error preview is displayed as shown below:

<img width="1726" alt="Screenshot 2024-07-24 at 2 37 49 PM" src="https://github.com/user-attachments/assets/53214cad-5797-4256-b1a1-77a94abcc6f7">

Error preview when the A4A plugin is installed:

<img width="1726" alt="Screenshot 2024-07-24 at 2 38 03 PM" src="https://github.com/user-attachments/assets/20f8f2ec-53bf-4a64-9a3f-df7f172b109e">

5. Verify the error preview looks good on various screens:


<img width="586" alt="Screenshot 2024-07-24 at 2 39 06 PM" src="https://github.com/user-attachments/assets/f358308c-3ee7-4563-98f8-3ceb5a3b4b5f">

<img width="282" alt="Screenshot 2024-07-24 at 2 39 27 PM" src="https://github.com/user-attachments/assets/32c27fec-d214-43e0-9fc0-c8a7ca95802d">


## Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

yashwin committed:
Implement site error preview

yashwin commented on client/a8c-for-agencies/sections/sites/features/jetpack/jetpack-sites-dataviews.tsx:
```diff
@@ -435,6 +437,7 @@ export const JetpackSitesDataViews = ( {
 			isLoading,
 			openSitePreviewPane,
 			renderField,
+			isNotProduction,
```
Added the missing dependency array

yashwin commented on client/a8c-for-agencies/sections/sites/features/jetpack/jetpack-sites-dataviews.tsx:
```diff
@@ -76,15 +78,13 @@ export const JetpackSitesDataViews = ( {
+			} ) );
 		},
-		[ setDataViewsState ]
+		[ isNotProduction, setDataViewsState ]
```
Added the missing dependency array

madebynoam reviewed (approved):
Thank you, Yashwin! The screenshots lgtm. The only thing I'm noticing is that header doesn't appear to have a border. Is it just the screenshots?

yashwin committed:
Add header border

yashwin commented:
> Thank you, Yashwin! The screenshots lgtm. The only thing I'm noticing is that header doesn't appear to have a border. Is it just the screenshots?

Thanks for the review, @madebynoam.

I have updated the PR by adding the border. 

<img width="1050" alt="Screenshot 2024-07-12 at 3 59 45 PM" src="https://github.com/user-attachments/assets/dc02e94d-e4b0-481f-b834-13115bedb389">

cleacos commented:
The external icon seems a bit unalignment:

![image](https://github.com/user-attachments/assets/713ee81f-ed20-499c-a427-cc459e7bebcb)

cleacos reviewed (commented):
It works as expected. The code looks good.

Some points:

- The external icon seems unalignment (screenshot above)
- If the site has connectivity issues and it no longer exists, I'd like to have the `Remove site` option somewhere. Maybe we could add the `Remove site` button inside the Error Preview Pane view.

yashwin committed:
Fix icon alignment

yashwin committed:
Implement remove site in the error preview

yashwin commented:
Thanks for the review, @cleacos!

> The external icon seems unalignment (screenshot above) 

I think some last-minute changes caused this issue. Fixed it now: https://github.com/Automattic/wp-calypso/pull/92544/commits/d0205fb79aa0814b5b0774f4cbe6acd62df9b4cc

> If the site has connectivity issues and it no longer exists, I'd like to have the Remove site option somewhere. Maybe we could add the Remove site button inside the Error Preview Pane view.

Great point as the user won't be able to remove the site if there is an error. So, providing this option is a good improvement.

I have added this option: https://github.com/Automattic/wp-calypso/pull/92544/commits/5b7ad3fa303088f97ed5e4815bf2415acfa2c17b

<img width="1920" alt="Screenshot 2024-07-15 at 11 56 17 AM" src="https://github.com/user-attachments/assets/69fb0a99-f47d-412c-b137-f4797462a18a">

<img width="1920" alt="Screenshot 2024-07-15 at 11 56 25 AM" src="https://github.com/user-attachments/assets/b20d3c2a-710a-4b02-bd63-74e72f3d7a22">

@cleacos, @madebynoam: Could you please take another look at this?

cleacos commented:
Great! Now it works perfectly!

cleacos reviewed (approved):
Thank you for addressing the feedback:

LGTM!

madebynoam commented:
Lgtm @yashwin. Would it be possible to replace the filled icon with this one?
https://www.figma.com/design/e4tLacmlPuZV47l7901FEs/WordPress-Design-Library?node-id=94411-9170&t=jeWYHv6kE2JDPm1E-4

yashwin committed:
Use <Icon/> instead of gridicons

yashwin commented:
Thanks for the review, @madebynoam!

> Would it be possible to replace the filled icon with this one?

Yes, I have updated all the icons. Now, using WP icons instead of Gridicons: https://github.com/Automattic/wp-calypso/pull/92544/commits/a39d09450702c454be8041360babd127df842935

yashwin committed:
Update KB article link for A4A

madebynoam commented:
> Thanks for the review, @madebynoam!
> 
> > Would it be possible to replace the filled icon with this one?
> 
> Yes, I have updated all the icons. Now, using WP icons instead of Gridicons: [a39d094](https://github.com/Automattic/wp-calypso/commit/a39d09450702c454be8041360babd127df842935)

Thank you, @yashwin. Lgtm!

a8ci18n commented:
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/16419856

Some locales (Hebrew, Japanese) have been temporarily machine-translated due to translator availability. All other translations are usually ready within a few days. Untranslated and machine-translated strings will be sent for translation next Monday and are expected to be completed by the following Friday.

Thank you @yashwin for including a screenshot in the description! This is really helpful for our translators.

a8ci18n commented:
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/16419856).
